src/missions/use-after-free-control-flow/README.md

```diff
@@ -1,3 +1,123 @@
+may be selected by the `cursor` of its owning `struct farm`. The crop at the
+cursor, if present, can be asked to describe itself, using a function pointer
+within the `struct crop` itself.
+
+To make matters more exciting, a UFO may be summoned to perform particular
+kinds of mischief. The UFO may "abduct" a pointer into itself and can create
+crop signs within `struct crop` (with data) or `struct farm` (with a capability
+to the `success()` function).
+
+The gadgets offered by the UFO are somewhat limited (that is, they are not
+"write what where" or arbitrary control transfer) as one might expect to see in
+a real application. Nevertheless, they are sufficiently powerful to provide
+*myriad* exploitation vectors on RISC-V without CHERI, and even a few on
+CHERI-RISC-V. However, we believe that enforced heap temporal safety will
+ensure that any use of them is either overwritten by subsequent program
+operations before the gadgets' effects influence control flow or will cause the
+program to fail-stop with a default-fatal signal (e.g., `SIGSEGV` or
+`SIGPROT`).
+
+### Building and running
+
+Using the `ccc` tool, we can build this program in three different
+configurations:
+* RISC-V without CHERI: `./ccc riscv64`
+* CHERI-RISC-V without heap temporal safety: `./ccc riscv64-purecap`
+* CHERI-RISC-V with temporal safety: `./ccc riscv64-purecap -DCAPREVOKE`
+
+**The latter of these is the only one relevant to the mission itself,** but
+exploring the other two may shed light on the program's operation.
+
+As said before, the program expects to read a command language from `stdin`.
+It will print "Ready" and then await input.  Unlike real applications, this
+program is fairly chatty about its own operation to ease exploration.
+Nevertheless, it may be useful to run it within `gdb`, especially to
+differentiate causes of crashes.  By way of example, here is a short session
+which creates one `struct farm`, two `struct crop`s, exhibits cursor control
+and description of `struct crop`s, and then tears down the `struct farm`:
+```
+$ echo FCC RD ZLD f | ./temporal-mission
+Ready (CHERI-RISC-V & heap safety)
+New farm (index 0) at 0x41201080
+New crop at 0x41201100
+New crop at 0x41201180
+Farm 0 cursor 0x41201180
+FYI: Current farm is 0x41201080
+FYI:  cursor 0x41201180, descr 0x102ebc
+Chelan at 0x41201180
+Farm 0 cursor 0x41201100
+FYI: Current farm is 0x41201080
+FYI:  cursor 0x41201100, descr 0x102ef8
+Colt at 0x41201100
+Tear down farm (index 0) at 0x41201080
+```
+
+### Command Language Directives
+
+* Whitespace is quietly ignored in most cases; this may simplify reading
+  programs.
+
+* Digits `0` through `3` focus on the corresponding farm slots, making it the
+  locus of subsequent commands until altered.
+
+* `F` allocates a new farm at the current slot.
+
+* `f` frees the current slot's farm (along with any crops in its collection)
+
+* `C` creates a new crop in the current farm and puts it at the left of the
+  collection.
```

Comment:
This needs to clarify how this relates to the cursor.

Reply:
Done, I think.